### PR TITLE
add deliver translations endpoint (WP-827)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

**Review by PavelLoparev — COMMENTED** [review]
Didn't we discuss that everything related to new rest api would be placed in the new branch and all PRs would be agains that new branch but not against master?

File: inc/Smartling/API/SettingsServiceApi.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        parent::__construct($projectId, self::initializeHttpClient(self::ENDPOINT_URL), $this->getLogger(), self::ENDPOINT_URL);
+        $this->setAuth($auth);
+    }
+
+    public function getSettings(): array
+    {
+        return $this->sendRequest('/integrations/wordpress/settings', $this->getDefaultRequestData('query', []), self::HTTP_METHOD_GET);
+    }
+}
```
Comment (PavelLoparev) [inline inc/Smartling/API/SettingsServiceApi.php:25]: Seems like it's time to update php sdk? But also, seems like it's time to split it to public and private parts?

File: inc/Smartling/API/SubmissionServiceApi.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Smartling\API;
+
+use Smartling\Models\AssetUid;
+use Smartling\Vendor\Smartling\AuthApi\AuthApiInterface;
+use Smartling\Vendor\Smartling\BaseApiAbstract;
+
+class SubmissionServiceApi extends BaseApiAbstract {
```
Comment (PavelLoparev) [inline inc/Smartling/API/SubmissionServiceApi.php:9]: It's already in the php sdk. I think you need to use service from sdk and wordpress related code should be placed in api wrapper.

File: inc/Smartling/ApiWrapperWithRetries.php [hunk 3]
```diff
@@ -60,6 +61,13 @@ public function downloadFile(SubmissionEntity $entity): string
         });
     }
 
+    public function getSettings(string $projectUid): array
+    {
+        return $this->withRetry(function () use ($projectUid) {
```
Comment (PavelLoparev) [inline inc/Smartling/ApiWrapperWithRetries.php:66]: is it possible to write `withRetry` as an annotation?

File: inc/Smartling/Models/Settings.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public function __construct(
+        private string $userIdentifier,
+        private string $secretKey,
+        private string $retrievalType,
+        private string $sourceLocale,
+        private int $sourceBlogId,
+        private array $targetLocales,
+    ) {
+    }
```
Comment (PavelLoparev) [inline inc/Smartling/Models/Settings.php:15]: Is this the only settings we would like to store in settings service? What about all the regexps and others?

**Comment by vsolovei-smartling** [comment]
> Didn't we discuss that everything related to the new rest API would be placed in the new branch and all PRs would be against that new branch but not against the master?

Good point, thanks, fixed

File: inc/Smartling/API/SettingsServiceApi.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/API/SettingsServiceApi.php:25]: Yes, I don't yet want to update the SDKs until we agree on what should go where

File: inc/Smartling/ApiWrapperWithRetries.php (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/Smartling/ApiWrapperWithRetries.php:66]: Yes, it should be, created https://bt.smartling.net/browse/WP-828 to research further and/or implement

**Review by dimitrystd — COMMENTED** [review]
Didn't finish review

**Pull request merged**